vm/trace.go

```diff
@@ -199,6 +199,8 @@ type ExecuteInvocation struct {
 }
 
 func (e ExecuteInvocation) MarshalJSON() ([]byte, error) {
+	// in this case RevertReason will be ignored, right?
+	// I don't remember exactly but it's either RevertReason or FunctionInvocation data?
```

Comment:
Yeap, it is essentially a union, it is either a revert reason or function invocation.